pkg/sentry/devices/nvproxy/uvm_mmap.go

```diff
@@ -29,8 +29,8 @@ func (fd *uvmFD) ConfigureMMap(ctx context.Context, opts *memmap.MMapOpts) error
 	// UVM_VALIDATE_VA_RANGE, and probably other ioctls, expect that
 	// application mmaps of /dev/nvidia-uvm are immediately visible to the
 	// driver.
-	if opts.PlatformEffect < memmap.PlatformEffectPopulate {
-		opts.PlatformEffect = memmap.PlatformEffectPopulate
+	if opts.PlatformEffect < memmap.PlatformEffectCommit {
+		opts.PlatformEffect = memmap.PlatformEffectCommit
```

Comment:
IIUC, `PlatformEffectCommit` is equivalent to mmap(MAP_POPULATE). As a side effect, it [prevents the MM from "overmapping"](https://cs.opensource.google/gvisor/gvisor/+/master:pkg/sentry/mm/address_space.go;l=178-181;drc=d8aa09e04c4e38155dfcee6ed0495c2b29f604fc). For UVM, like you found out, overmapping (with wrong offset etc) can cause failures, so this has the right effect.

@nixprime Does this look good to you?

Reply:
@derpsteb Can you confirm if https://github.com/google/gvisor/pull/10008 fixes the issue without this change?

Reply:
Nice. I tested #10008 without the above change and that works fine. I will thus drop [bfe2eb7ab838c73767a0906a1b509bc67f75e276](https://github.com/google/gvisor/pull/9997/commits/bfe2eb7ab838c73767a0906a1b509bc67f75e276) from this PR.